tests/integrations/client/client_test.go

```diff
@@ -222,6 +222,21 @@ func TestLeaderTransfer(t *testing.T) {
 		newLeaderName := cluster.WaitLeader()
 		re.NotEqual(oldLeaderName, newLeaderName)
 	}
+
+	// ABC->ABCDEF
+	oldServers := cluster.GetServers()
+	for i := 0; i < 3; i++ {
+		cluster.Join(ctx)
+	}
+	oldLeaderName := cluster.WaitLeader()
+
+	// ABCDEF->DEF
+	for _, s := range oldServers {
+		s.Stop()
+	}
+	newLeaderName := cluster.WaitLeader()
+	re.NotEqual(oldLeaderName, newLeaderName)
```

Comment:
Should we do more checks to ensure the new leader comes from the DEF rather than ABC here?